### PR TITLE
fix: bootnodes

### PR DESCRIPTION
**Pull request by PhilippLgh**
updates bootstrap node according to
https://github.com/ethereum/go-ethereum/blob/master/params/bootnodes.go
https://github.com/openethereum/openethereum/blob/master/ethcore/res/ethereum/kovan.json

adds a script to semi-automate process

### CODE REVIEW

File: packages/common/src/chains/mainnet.json [hunk 1]
```diff
@@ -77,32 +77,60 @@
   ],
   "bootstrapNodes": [
     {
-      "ip": "13.93.211.84",
-      "port": 30303,
-      "id": "3f1d12044546b76342d59d4a05532c14b85aa669704bfe1f864fe079415aa2c02d743e03218e57a33fb94523adb54032871a6c51b2cc5514cb7c7e35b3ed0a99",
-      "location": "US-WEST",
-      "comment": "Go Bootnode"
+      "ip": "18.138.108.67",
+      "port": "30303",
```
Comment (holgerd77) [inline packages/common/src/chains/mainnet.json:81]: To change the type for `port` would be a breaking change for the library, please don't side-introduce here.

Comment (PhilippLgh) [reply packages/common/src/chains/mainnet.json:81]: good point as this is not covered by any test 👍

Comment (PhilippLgh) [reply packages/common/src/chains/mainnet.json:81]: actually it kind of was and I "fixed" it :D

Comment (holgerd77) [reply packages/common/src/chains/mainnet.json:81]: 😛

**Comment by PhilippLgh** [comment]
@holgerd77 I was using a script to transform between our json representation and the one used by geth. added it as it could be one step in the direction of a more automated and less error prone approach. please note that we are not using a uniform structure: port was type number in ropsten json (not covered by test)

**Comment by holgerd77** [comment]
Blocked by https://github.com/ethereumjs/ethereumjs-devp2p/issues/59

File: packages/common/tests/chains.ts [hunk 2]
```diff
@@ -59,8 +59,7 @@ tape('[Common]: Initialization / Chain params', function(t: tape.Test) {
     const hash = '0xd4e56740f876aef8c010b86a40d5f56745a118d0906a34e69aec8c0db1cb8fa3'
     st.equal(c.genesis().hash, hash, 'should return correct genesis hash')
     st.equal(c.hardforks()[3]['block'], 2463000, 'should return correct hardfork data')
-    st.equal(c.bootstrapNodes()[0].port, 30303, 'should return a bootstrap node array')
-
+    st.equal(typeof c.bootstrapNodes()[0].port, 'number', 'should return a bootstrap node array')
```
[the remark this answers is not in the document]
Comment (PhilippLgh) [reply packages/common/tests/chains.ts:62]: I think the message reflects the intention of the test which is that the object includes bootnodes. it tests it by checking the port which is weird but works. Maybe the cleaner approach is to leave this test and write another one that tests the structural integrity of the bootnodes object which would help with problems like: https://github.com/ethereumjs/ethereumjs-devp2p/issues/59

**Comment by PhilippLgh** [comment]
added a test for the bootnode structure. another idea is of course to let typescript handle this task for us and not use json files

**Comment by evertonfraga** [comment]
@holgerd77 Yes. I will prepare the major release of Common, along with the other packages. 

Those are ready to go:
- Integrating Block v3.0.0 into Blockchain
- Integrating Blockchain into VM
- Scoped packages (tons of changes ranging from README to code references and documentation)

I am almost done with the `Block > VM` integration, and as soon as I finish it, I will raise the PRs.

**Comment by holgerd77** [comment]
Great, thanks

**Comment by evertonfraga** [comment]
@PhilippLgh instead of migrating the data to TypeScript, which makes the data less portable, we can put a JSON schema in place defining the bootnodes format, and validate it using [AJV](https://www.npmjs.com/package/ajv), for example. 

Eventually, we could use another JSON schema for the Genesis files as well.

**Comment by holgerd77** [comment]
Would also support that this should remain `JSON` for portability reasons, yes, validation using a proven tool might be a good idea to add.

**Pull request merged**